### PR TITLE
Estiaan/additional documents

### PR DESCRIPTION
**Pull request by madwizzard**
I will add my reports when the unit tests are done 

image

### CODE REVIEW

File: RR.UnitOfWork/RR.UnitOfWork.csproj [hunk 1]
```diff
@@ -31,7 +31,6 @@
   <ItemGroup>
     <Folder Include="Entities\ATS\" />
     <Folder Include="Interfaces\ATS\" />
-    <Folder Include="Migrations\" />
```
Comment (tbddorothy) [inline RR.UnitOfWork/RR.UnitOfWork.csproj:34]: Why did you delete migrations????

Comment (madwizzard) [reply RR.UnitOfWork/RR.UnitOfWork.csproj:34]: I did not this happens when i run add-migration but I will try it again

File: RR.App.Tests/Controllers/HRIS/EmployeeDocumentControllerUnitTest.cs [hunk 2]
```diff
@@ -16,7 +16,6 @@ namespace RR.App.Tests.Controllers.HRIS;
 public class EmployeeDocumentControllerUnitTest
 {
     private readonly Mock<IEmployeeDocumentService> _employeeMockDocumentService;
-    //private readonly EmployeeDocumentDto _employeeDocument;
```
Comment (Juanro-Minne) [inline RR.App.Tests/Controllers/HRIS/EmployeeDocumentControllerUnitTest.cs:19]: comment

**Review by Juanro-Minne — DISMISSED** [review]

**Review by MathildaShongwe15 — DISMISSED** [review]

**Comment by tbddorothy** [comment]
@EstiaanBritz @madwizzard You need to add back the migrations please

File: RR.App/Controllers/HRIS/EmployeeDocumentController.cs [hunk 3]
```diff
@@ -18,13 +18,13 @@ public EmployeeDocumentController(IEmployeeDocumentService employeeDocumentServi
     }
 
     [Authorize(Policy = "AllRolesPolicy")]
-    [HttpGet("{employeeId}")]
-    public async Task<IActionResult> GetAllEmployeeDocuments(int employeeId)
+    [HttpGet("all/{employeeId}/{documentType}")]
```
Comment (tbddorothy) [inline RR.App/Controllers/HRIS/EmployeeDocumentController.cs:21]: This all is not necessary in this endpoint name

File: RR.App/Controllers/HRIS/EmployeeDocumentController.cs [hunk 4]
```diff
@@ -18,13 +18,13 @@ public EmployeeDocumentController(IEmployeeDocumentService employeeDocumentServi
...
+    public async Task<IActionResult> GetAllEmployeeDocuments(int employeeId, int documentType)
     {
         try
         {
-            var employeeDocuments = await _employeeDocumentService.GetAllEmployeeDocuments(employeeId);
-            return Ok(employeeDocuments);
+                var employeeDocuments = await _employeeDocumentService.GetAllEmployeeDocuments(employeeId, (DocumentType)documentType);
```
Comment (tbddorothy) [inline RR.App/Controllers/HRIS/EmployeeDocumentController.cs:26]: You can just say GetEmployeeDocuments and the parameters show that it's by employeeId and documentType

File: HRIS.Models/Enums/DocumentType.cs [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace HRIS.Models.Enums;
+
+public enum DocumentType
+{
+    starterKit,
```
Comment (tbddorothy) [inline HRIS.Models/Enums/DocumentType.cs:11]: Please use PascalCase for Enums

File: HRIS.Services.Tests/Services/EmployeeDocumentServiceUnitTest.cs [hunk 6]
```diff
@@ -83,14 +98,164 @@ public async Task SaveEmployeeDocumentPass()
...
+            .addNewAdditionalDocument(employeeDocDto, email, documentType));
+
+        Assert.Equal("employee not found", exception.Message);
+
+        _errorLoggingServiceMock.Verify(x => x.LogException(It.IsAny<Exception>()), Times.Once);
+    }
+
+
+    //[Theory]
+    //[InlineData(true, true, 0, DocumentStatus.PendingApproval, DocumentType.starterKit, "test@retrorabbit.co.za", 1)]
```
Comment (tbddorothy) [inline HRIS.Services.Tests/Services/EmployeeDocumentServiceUnitTest.cs:185]: And then?

File: RR.App/Controllers/HRIS/EmployeeDocumentController.cs (hunk 3 above)
Comment (tbddorothy) [reply RR.App/Controllers/HRIS/EmployeeDocumentController.cs:21]: Based on our chat I'm resolving

**Review by tbddorothy — APPROVED** [review]

**Review by LlewellynAnthony — APPROVED** [review]

**Review by dvanhout89 — APPROVED** [review]
